packages/gui/src/components/inputs/Dimension/Input.tsx

```diff
@@ -151,7 +156,7 @@ export function DimensionInput<K extends string = never>(
               onChange({ value: unitValue, unit: newUnit })
             } else if (newUnit === 'theme') {
               const themeValue = themeValues?.[0]
-              onChange(themeValue)
+              onChange({ ...themeValue, themePath: `${themeProperty}.[${0}]` })
```

Comment:
I think we want to use Theme UI's `get` function to keep things consistent since its pathing would be `${themeProperty}.0` rather than the `[]` syntax being there.